### DISCUSSION TITLE
Model stop and report error "Unexpected Termination, reason unknown"

### DISCUSSION THREAD

**Discussion by wanyingw193** [NREL/ReEDS-2.0#78] [Q&A: Execution]
Hi,

The model stop and came out the following error. What would be the suggestion to solve this error? Thanks!

1638146 Equation   eq_water_use_limit    0.000   217.312 SECS 25,814 MB      0
                                      0.000              ROWGEN
                                      0.000              COLGEN
*** executing line 1638146

*** Unexpected Termination, reason unknown
*** Please report to support@gams.com
*** including version ( GAMS 30.3.0 ) and platform ( WEI )

**Comment by vcelloho** [comment]
There are some compatibility issues with the water use calculations where certain switch settings conflict. Were you intending to run with water consumption information?

**Reply by wanyingw193** [reply]
Hi, the water consumption information is not the focus of the run. Would it be appropriate to release some water consumption constraints in order to get the model to run? Thanks!

**Comment by vcelloho** [comment]
Likely the culprit is if you have GSw_Upgrades enabled this conflicts with, the water constraints. Check that the following switches are set to zeros.

GSw_WaterCapacity
GSw_WaterMain	
GSw_WaterUse

**Reply by wanyingw193** [reply]
Hi, it is very strange that I have set all three switches to 0. The eq_water_use_limit should not be called, but it still runs into this error?

**Reply by vcelloho** [reply]
Hmm can you post the code for the line in your version? Which point in the execution does this crash occur in, which .lst file is generated last.

**Reply by wanyingw193** [reply]
Sure! It stop at executing eq_water_use_limit in 2028, I didnt change the code for eq_water_use_limit, the code and .lst file are attached.

===========================code==================================
`eq_water_use_limit(i,v,w,r,szn,t)$[i_water_cooling(i)$valgen(i,v,r,t)$tmodel(t)$i_w(i,w)$Sw_WaterMain$Sw_WaterCapacity$Sw_WaterUse]..

    WATCAP(i,v,r,t) *sum{wst$i_wst(i,wst), watsa(wst,r,szn,t) }

    =g=

    sum{h$h_szn(h,szn), WAT(i,v,w,r,h,t) }
;`


===========================2028.lst============================
GAMS 30.3.0  rc5da09e Released Mar  6, 2020 WEX-WEI x86 64bit/MS Windows                                                                                                                                                             12/07/21 02:39:16 Page 342
----1638042 Equation   eq_storage_thermalres    0.000   230.547 SECS 25,770 MB      0
'ReEDS 2.0'
Model Analysis      SOLVE cws_Acap0RelLhcstwu Using LP From line 1641788


                                      0.000              ROWGEN
                                      0.000              COLGEN
----1638022 Equation   eq_storage_level    2.359   232.906 SECS 25,802 MB  123182
                                      2.125              ROWGEN
                                      0.234              COLGEN
----1638067 Equation   eq_storage_in_min    0.328   233.234 SECS 25,802 MB   2278
                                      0.188              ROWGEN
                                      0.140              COLGEN
----1638078 Equation   eq_storage_in_max    1.813   235.047 SECS 25,813 MB   6834
                                      1.453              ROWGEN
                                      0.360              COLGEN
----1638099 Equation   eq_Canadian_Imports    0.000   235.047 SECS 25,813 MB     36
                                      0.000              ROWGEN
                                      0.000              COLGEN
----1638113 Equation   eq_water_accounting    0.015   235.062 SECS 25,813 MB      0
                                      0.015              ROWGEN
                                      0.000              COLGEN
----1638124 Equation   eq_water_capacity_total    0.000   235.062 SECS 25,813 MB      0
                                      0.000              ROWGEN
                                      0.000              COLGEN
----1638134 Equation   eq_water_capacity_limit    0.000   235.062 SECS 25,813 MB      0
                                      0.000              ROWGEN
                                      0.000              COLGEN
----1638146 Equation   eq_water_use_limit    0.000   235.062 SECS 25,813 MB      0
                                      0.000              ROWGEN
                                      0.000              COLGEN
*** executing line 1638146

*** Unexpected Termination, reason unknown
*** Please report to support@gams.com
*** including version ( GAMS 30.3.0 ) and platform ( WEI )

**Comment by wanyingw193** [comment]
Hi, if there is an unexpected termination of the model, is there any action that can be taken to restart the model from the termination step and continue the unfinished run?

Thank you!
